# **Netflix Movies and TV Shows - Data Cleaning**

## **Часть 1. Введение**

Данный датасет содержит информацию о фильмах и сериалах, доступных на Netflix.
В сыром виде данные содержат пропуски в колонках, смешанные форматы и даты, которые хранятся в виде строк.

Целью проекта является провести осмысленную очистку данных: разобраться в природе пропусков по каждой колонке и принять обоснованное решение по ней, привести типы данных к корректным, и подготовить датасет к дальнейшему анализу.

## **Часть 2. Загрузка и первый обзор данных**

In [1]:
import pandas as pd
import os
from pprint import pprint

In [2]:
os.chdir('..') 
df = pd.read_csv('data/raw/netflix_titles.csv')

In [3]:
print(f"Датафрейм содержит {df.shape[0]} строк и {df.shape[1]} колонок")

Датафрейм содержит 8807 строк и 12 колонок


In [4]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   show_id       8807 non-null   str  
 1   type          8807 non-null   str  
 2   title         8807 non-null   str  
 3   director      6173 non-null   str  
 4   cast          7982 non-null   str  
 5   country       7976 non-null   str  
 6   date_added    8797 non-null   str  
 7   release_year  8807 non-null   int64
 8   rating        8803 non-null   str  
 9   duration      8804 non-null   str  
 10  listed_in     8807 non-null   str  
 11  description   8807 non-null   str  
dtypes: int64(1), str(11)
memory usage: 825.8 KB


In [5]:
missing = df.isna().sum()  # количество пропущенных значений по каждому столбцу
missing.sort_values(ascending=False)

director        2634
country          831
cast             825
date_added        10
rating             4
duration           3
show_id            0
type               0
title              0
release_year       0
listed_in          0
description        0
dtype: int64

In [6]:
round(missing / len(df) * 100, 2).sort_values(ascending=False)  # процент пропусков по каждому столбцу

director        29.91
country          9.44
cast             9.37
date_added       0.11
rating           0.05
duration         0.03
show_id          0.00
type             0.00
title            0.00
release_year     0.00
listed_in        0.00
description      0.00
dtype: float64

In [7]:
if df.duplicated().sum() == 0:
    print("Полностью совпадающих строк нет")
else:
    print(f"Найдено {len(df[df.duplicated()])} дубликатов")
    df[df.duplicated()]

Полностью совпадающих строк нет


In [8]:
df.head(10)

,show_id,type,title,director,cast,country,date_added,release_year,rating,duration,listed_in,description
0,s1,Movie,Dick Johnson Is Dead,Kirsten Johnson,NaN,United States,"September 25, 2021",2020,PG-13,90 min,Documentaries,"As her father nears the end of his life, filmm..."
1,s2,TV Show,Blood & Water,NaN,"Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...",South Africa,"September 24, 2021",2021,TV-MA,2 Seasons,"International TV Shows, TV Dramas, TV Mysteries","After crossing paths at a party, a Cape Town t..."
2,s3,TV Show,Ganglands,Julien Leclercq,"Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...",NaN,"September 24, 2021",2021,TV-MA,1 Season,"Crime TV Shows, International TV Shows, TV Act...",To protect his family from a powerful drug lor...
3,s4,TV Show,Jailbirds New Orleans,NaN,NaN,NaN,"September 24, 2021",2021,TV-MA,1 Season,"Docuseries, Reality TV","Feuds, flirtations and toilet talk go down amo..."
4,s5,TV Show,Kota Factory,NaN,"Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...",India,"September 24, 2021",2021,TV-MA,2 Seasons,"International TV Shows, Romantic TV Shows, TV ...",In a city of coaching centers known to train I...
5,s6,TV Show,Midnight Mass,Mike Flanagan,"Kate Siegel, Zach Gilford, Hamish Linklater, H...",NaN,"September 24, 2021",2021,TV-MA,1 Season,"TV Dramas, TV Horror, TV Mysteries",The arrival of a charismatic young priest brin...
6,s7,Movie,My Little Pony: A New Generation,"Robert Cullen, José Luis Ucha","Vanessa Hudgens, Kimiko Glenn, James Marsden, ...",NaN,"September 24, 2021",2021,PG,91 min,Children & Family Movies,Equestria's divided. But a bright-eyed hero be...
7,s8,Movie,Sankofa,Haile Gerima,"Kofi Ghanaba, Oyafunmike Ogunlano, Alexandra D...","United States, Ghana, Burkina Faso, United Kin...","September 24, 2021",1993,TV-MA,125 min,"Dramas, Independent Movies, International Movies","On a photo shoot in Ghana, an American model s..."
8,s9,TV Show,The Great British Baking Show,Andy Devonshire,"Mel Giedroyc, Sue Perkins, Mary Berry, Paul Ho...",United Kingdom,"September 24, 2021",2021,TV-14,9 Seasons,"British TV Shows, Reality TV",A talented batch of amateur bakers face off in...
9,s10,Movie,The Starling,Theodore Melfi,"Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...",United States,"September 24, 2021",2021,PG-13,104 min,"Comedies, Dramas",A woman adjusting to life after a loss contend...


In [9]:
df.describe()

,release_year
count,8807.000000
mean,2014.180198
std,8.819312
min,1925.000000
25%,2013.000000
50%,2017.000000
75%,2019.000000
max,2021.000000


In [10]:
columns = df.columns
columns

Index(['show_id', 'type', 'title', 'director', 'cast', 'country', 'date_added',
       'release_year', 'rating', 'duration', 'listed_in', 'description'],
      dtype='str')

In [ ]:
# Количество уникальных значений по столбцам
for column in columns:
    print(f"{column}: {df[column].nunique()}")

Количество уникальных значений:
show_id: 8807
type: 2
title: 8807
director: 4528
cast: 7692
country: 748
date_added: 1767
release_year: 74
rating: 17
duration: 220
listed_in: 514
description: 8775


In [1]:
# Для просмотра уникальных значений нужно раскомментировать код

# Сами уникальные значения
# for column in columns:
#     pprint(f"{column}: {df[column].unique().tolist()}")
#     print('\n')

### **Выводы:**

В таблице всего 8807 строк и 12 столбцов. Полностью совпадающих строк нет.

Пропуски присутствуют в 6 колонках: в колонке `director` 2634 пропусков, что составляет 29.91% (почти треть столбца), в `country` - 831 (9.44%), `cast` - 825 (9.37%), `date_added` - 10 (0.11%), `rating` - 4 (0.05%), `duration` - 3 (0.03%). В столбцах без пропусков никаких аномалий нет.

